src/bindgen/ir/enumeration.rs

```diff
@@ -700,28 +700,24 @@ impl Source for Enum {
                         for &(ref field_name, ref ty, ..) in body.fields.iter().skip(skip_fields) {
                             out.new_line();
                             match ty {
-                                Type::Array(ref _ty, ref length) => {
+                                Type::Array(ref ty, ref length) => {
                                     // arrays are not assignable in C++ so we
                                     // need to manually copy the elements
+                                    write!(out, "for (int i = 0; i < {}; i++)", length.as_str());
+                                    out.open_brace();
                                     write!(
                                         out,
-                                        "for (int i = 0; i < {}; i++) {{\
-                                         result.{}.{}[i] = {}[i];\
-                                         }}",
-                                        length.as_str(),
-                                        variant_name,
-                                        field_name,
-                                        arg_renamer(field_name)
+                                        "new (result.{}.{} + i) ",
```

Comment:
As we want to be sure to use the global new operator here (provided by `<new>`), it's probably worthwhile to generate the leading `::` to force scope resolution to the global operator.

I also generally prefer avoiding using array->pointer implicit conversions like those here, so imo it might be nicer to write `&result.{}.{}[i]`.

Reply:
Fair :)